java/maven.hints/src/org/netbeans/modules/maven/hints/errors/EnablePreviewMavenProj.java

```diff
@@ -246,47 +226,176 @@ private Configuration createConfiguration() {
         }
 
         private Plugin updateMavenCompilerPlugin(final Plugin oldPlugin) {
-
             Configuration currenConfig = oldPlugin.getConfiguration();
+            Plugin newPlugin = factory.createPlugin();
+            newPlugin.setGroupId(oldPlugin.getGroupId());
+            newPlugin.setArtifactId(oldPlugin.getArtifactId());
+            newPlugin.setVersion(oldPlugin.getVersion());
+            newPlugin.setConfiguration(updateMavenCompilerPluginConfiguration(currenConfig, oldPlugin.getVersion()));
+            return newPlugin;
+        }
+
+        private Configuration updateMavenCompilerPluginConfiguration(Configuration currenConfig, String version) {
             Configuration newConfiguration = createConfiguration();
 
-            boolean isCompilerArgsElementPresent = false;
-            if (currenConfig != null) {
-                for (POMExtensibilityElement element : currenConfig.getConfigurationElements()) {
-                    POMExtensibilityElement newElement = factory.createPOMExtensibilityElement(element.getQName());
-                    String elementText = element.getElementText();
-                    if (elementText.trim().length() > 0) {
-                        newElement.setElementText(element.getElementText());
+            if (currenConfig == null) {
+                currenConfig = createConfiguration();
+            }
+
+            boolean supportsRelease = false;
+
+            try {
+                supportsRelease = version == null ||
+                                  new SpecificationVersion(version).compareTo(new SpecificationVersion("3.6")) >= 0;
```

Comment:
consider using `ComparableVersion`. 
- this won't throw NumberFormatExceptions
- and it will also support text like `3.10-SNAPSHOT`

Reply:
added a commit for this: https://github.com/mbien/netbeans/commits/better-enable-preview